### PR TITLE
SDV converter

### PR DESCRIPTION
**Pull request by michaelcfanning**
Converter for static driver verifier. @lgolding @rtaket

### CODE REVIEW

File: src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif [hunk 1]
```diff
@@ -0,0 +1,2122 @@
+{
+  "$schema": "http://json.schemastore.org/sarif-1.0.0",
+  "version": "1.0.0",
+  "runs": [
+    {
+      "tool": {
+        "name": "StaticDriverVersion"
```
Comment (lgolding) [inline src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif]: "StaticDriverVerifier"

File: src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif [hunk 2]
```diff
@@ -0,0 +1,2122 @@
...
+      },
+      "files": {
+        "..\\..\\..\\rules\\WDM\\CheckAddDevice.slic": {
```
Comment (lgolding) [inline src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif]: These don't look like C++ files. They have their own syntax.

File: src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif (hunk 1 above)
Comment (michaelcfanning) [inline src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif]: Ouch. yes.

File: src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif (hunk 2 above)
Comment (michaelcfanning) [inline src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif]: Agreed, will remove for now.

File: src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif [hunk 3]
```diff
@@ -0,0 +1,2122 @@
...
+                  }
+                },
+                {
+                  "id": 3,
+                  "message": "====Auto=====,====Pre=====,____true",
+                  "kind": "call",
+                  "essential": false,
+                  "properties": {
+                    "Step": 2,
+                    "State": "^====Auto=====^====Pre=====^____true",
```
Comment (lgolding) [inline src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif]: It looks like "State" is always the same as "Message". Do we need both?

File: src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/src/fail_driver1.c [hunk 4]
```diff
@@ -0,0 +1,253 @@
+/*++
+
+Copyright (c) Microsoft Corporation.  All rights reserved.
```
Comment (lgolding) [inline src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/src/fail_driver1.c]: Is it ok for all these files under StaticDriverVerifier/src to be in a public repo?

**Comment by lgolding** [comment]
CodeFlow isn't letting me review this effectively. It isn't showing me the diffs for any of the files under the Sarif project. Going to switch back to GitHub and see if I can see it there.

File: src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif (hunk 3 above)
Comment (michaelcfanning) [inline src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/checkadddevice_bug1.tt.sarif]: I actually convert the state ^ to a comma and delete the leading ^. We need a better message. Will address this once we look at the actual SDV viewer.

File: src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/src/fail_driver1.c (hunk 4 above)
Comment (michaelcfanning) [inline src/Sarif.FunctionalTests/ConverterTestData/StaticDriverVerifier/src/fail_driver1.c]: yes it is, verified with SDV team

**Pull request merged**